### PR TITLE
set maven libs according to consumers

### PR DESCRIPTION
**Pull request by jtnord**
use a version of maven libs that we expect to be satisfied by consumers.

cf https://github.com/jenkinsci/maven-hpi-plugin/pull/440#issuecomment-1454825697

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jtnord** [comment]
requires maven-jellydoc-plugin#34

File: pom.xml [hunk 1]
```diff
@@ -131,6 +135,12 @@
         <artifactId>plexus-utils</artifactId>
         <version>3.5.0</version>
       </dependency>
+      <dependency>
+        <!-- upper bounds conflict from doxia-integration via maven-reporting-impl via maven-jellydoc-plugin -->
```
Comment (jtnord) [inline pom.xml:139]:
```
Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (display-info) on project maven-stapler-plugin:
Rule 0: org.apache.maven.enforcer.rules.dependency.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.codehaus.plexus:plexus-interpolation:1.25 paths to dependency are:
+-org.kohsuke.stapler:maven-stapler-plugin:1.23-SNAPSHOT
  +-org.apache.maven:maven-core:3.8.1 [provided]
    +-org.apache.maven:maven-settings-builder:3.8.1 [provided]
      +-org.codehaus.plexus:plexus-interpolation:1.25
and
+-org.kohsuke.stapler:maven-stapler-plugin:1.23-SNAPSHOT
  +-org.apache.maven:maven-core:3.8.1 [provided]
    +-org.apache.maven:maven-model-builder:3.8.1 [provided]
      +-org.codehaus.plexus:plexus-interpolation:1.25 [provided]
and
+-org.kohsuke.stapler:maven-stapler-plugin:1.23-SNAPSHOT
  +-org.jvnet.maven-jellydoc-plugin:maven-jellydoc-plugin:1.11-rc107.33dc688a_2b_92
    +-org.apache.maven.reporting:maven-reporting-impl:3.2.0
      +-org.apache.maven.doxia:doxia-integration-tools:1.11.1
        +-org.codehaus.plexus:plexus-interpolation:1.26

```

File: pom.xml [hunk 2]
```diff
@@ -164,7 +174,7 @@
     <dependency>
       <groupId>org.jvnet.maven-jellydoc-plugin</groupId>
       <artifactId>maven-jellydoc-plugin</artifactId>
-      <version>1.10</version>
+      <version>1.11-rc107.33dc688a_2b_92</version>
```
Comment (jtnord) [inline pom.xml:177]: https://github.com/jenkinsci/maven-jellydoc-plugin/pull/34

**Review by basil — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (jtnord) [inline pom.xml:177]:
```suggestion
      <version>1.11</version>
```

**Pull request merged**